Google.Solutions.Compute/Net/PortFinder.cs

```diff
@@ -0,0 +1,68 @@
+﻿//
+// Copyright 2019 Google LLC
+//
+// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+// 
+//   http://www.apache.org/licenses/LICENSE-2.0
+// 
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+//
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Net.NetworkInformation;
+
+namespace Google.Solutions.Compute.Test.Net
+{
+    /// <summary>
+    /// Helper class to find unsed local TCP ports.
+    /// </summary>
+    internal static class PortFinder
```

Comment:
there sins't a c# function to 'get a free port()"'...i'd guess not (there's a good reason you have this in a  cllass)

Reply:
The only other way I am aware of is this:

```
               TcpListener listener = new TcpListener(IPAddress.Loopback, 0);
		try
		{
			listener.Start();
			return ((IPEndPoint)listener.LocalEndpoint).Port;
		}
		finally
		{
			listener.Stop();
		}
```

That's what `Google.Apis.Auth.OAuth2.LocalServerCodeReceiver` does. But this actually opens the port, so I did not like it too much.